### ISSUE TITLE
No such file or directory: '../../data_files/lpd_5/lpd_5_full/0/00000ec8a66b6bd2ef809b0443eeae41.npz

### ISSUE THREAD

**Issue by yc-li20** [serkansulun/midi-emotion#5]
[](url)Hi Serkan,
When I tried to run **preprocess_pianorolls.py**, it says No such file or directory: '../../data_files/lpd_5/lpd_5_full/0/00000ec8a66b6bd2ef809b0443eeae41.npz

Is it because the path should be **input_dir = "../../data_files/lpd_full/lpd/lpd_full"** instead of "../../data_files/lpd_5/lpd_5_full"?

Thank you.

![Screenshot 2023-08-22 175631](https://github.com/serkansulun/midi-emotion/assets/15671418/f07859da-fe63-4bd9-b7f7-9680f054d91c)
![Screenshot 2023-08-22 175646](https://github.com/serkansulun/midi-emotion/assets/15671418/34f045ea-1095-4f07-b37d-51dd7cab6c84)

**Comment by serkansulun** [comment]
Hi!

I noticed that in the readme, I linked the wrong dataset, now I fixed it. Indeed, you should download and use the lpd_5_full dataset because it only has 5 instruments and you need a limited amount of instruments so that you'll end up with a fixed amount of tokens.

Thanks for the notification!

**Comment by yc-li20** [comment]
Hi Serkan, thank you for your prompt response!

So, you mean I should use lpd_5_full instead of lpd_full, and change every path named **lpd_full to lpd_5_full**?

Also, could you please let me know how long does it take to finish training the model and how is it going to end?
I did some modification and am training a new one. Now it has been 15 hours, and it is 11685 epoch and the training loss is 3.5476, but the validation loss doesn't go down since the Eval 1 step. Does it affect the training?
Thank you.

**Comment by serkansulun** [comment]
Yes. By default, the paths should be lpd_5_full anyway. You just need to download the right dataset using the updated link in the readme.

I don't exactly remember how long it took, but it might have been a week on an RTX 6000 GPU (maybe a few million steps, not sure). The training is finished when the training loss converges (stabilizes). As I said in the paper, after this happened, I reduced the learning rate and kept training a bit more until another convergence.

Validation loss definitely should go down, something is wrong there.

Finally, I made pretrained models available (see readme) so you can use them without any training.

**Comment by yc-li20** [comment]
Thank you!!

**Comment by yc-li20** [comment]
Hi Serkan, I may have a follow up question.
Could you please guide me how did you load the trained model and reduce the lr for continue training? I noticed the overwrite_lr argument in the config, but am not sure how exactly did you do.
Thank you!

**Comment by serkansulun** [comment]
```
python train.py --restart_dir path/to/model_dir --overwrite_lr --lr 2e-6
```

Considering the trained models that I've shared, I already trained them further with a smaller learning rate so you don't have to do that if you're using my trained models.

**Comment by yc-li20** [comment]
Hi Serkan, sorry for bothering. I have one last question.
Is it possible that when generating music, we keep the music embeddings the same (same initialization for the transformer, maybe) and only change the emotion embeddings given different v and a value? I am curious what a same piece of music with different emotions will sound like.
I guess I should ensure the "gen_song_tensor" remain the same in generate.py. Am I right? Could you please give me some guidance if this practice is possible? Thank you so much!

**Comment by serkansulun** [comment]
The best you can do is to fix the first n tokens and use a positive integer for seed. You'll basically see how the model extends the same piece of melody, using different emotions. Yes, you can fix the `gen_song_tensor`, or the `primer`.  But you should only do this for the initialization, otherwise, you'd get the same output every time.
